### PR TITLE
[Docs] Switch steps 3 and 4 in M1 installation instructions

### LINKED ISSUE #35509: [M1 Installation] Ray cannot be imported after installation due to gRPC
### Description

After following [the Apple M1 installation directions](https://docs.ray.io/en/releases-2.4.0/ray-overview/installation.html#m1-mac-apple-silicon-support) on a fresh Conda environment, I was unable to import Ray:

```python
>>> import ray
Traceback (most recent call last):
  File "/Users/shrekris/miniforge3/envs/aehf/lib/python3.10/site-packages/ray/__init__.py", line 63, in _configure_system
    import grpc  # noqa: F401
  File "/Users/shrekris/miniforge3/envs/aehf/lib/python3.10/site-packages/grpc/__init__.py", line 22, in <module>
    from grpc import _compression
  File "/Users/shrekris/miniforge3/envs/aehf/lib/python3.10/site-packages/grpc/_compression.py", line 15, in <module>
    from grpc._cython import cygrpc
ImportError: dlopen(/Users/shrekris/miniforge3/envs/aehf/lib/python3.10/site-packages/grpc/_cython/cygrpc.cpython-310-darwin.so, 2): Symbol not found: _CFRelease
  Referenced from: /Users/shrekris/miniforge3/envs/aehf/lib/python3.10/site-packages/grpc/_cython/cygrpc.cpython-310-darwin.so
  Expected in: flat namespace
 in /Users/shrekris/miniforge3/envs/aehf/lib/python3.10/site-packages/grpc/_cython/cygrpc.cpython-310-darwin.so

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/shrekris/miniforge3/envs/aehf/lib/python3.10/site-packages/ray/__init__.py", line 101, in <module>
    _configure_system()
  File "/Users/shrekris/miniforge3/envs/aehf/lib/python3.10/site-packages/ray/__init__.py", line 65, in _configure_system
    raise ImportError(
ImportError: Failed to import grpc on Apple Silicon. On Apple Silicon machines, try `pip uninstall grpcio; conda install grpcio`. Check out https://docs.ray.io/en/master/ray-overview/installation.html#m1-mac-apple-silicon-support for more details.
```

I expect the instructions in the docs to include the `pip uninstall grpcio; conda install grpcio` instruction if it's necessary.

### Link

https://docs.ray.io/en/releases-2.4.0/ray-overview/installation.html#m1-mac-apple-silicon-support

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
## Why are these changes needed?

The M1 installation instructions do not install Ray successfully. They produce a gRPC-related error with a scary-looking output:

```python
>>> import ray
Traceback (most recent call last):
  File "/Users/shrekris/miniforge3/envs/aehf/lib/python3.10/site-packages/ray/__init__.py", line 63, in _configure_system
    import grpc  # noqa: F401
  File "/Users/shrekris/miniforge3/envs/aehf/lib/python3.10/site-packages/grpc/__init__.py", line 22, in <module>
    from grpc import _compression
  File "/Users/shrekris/miniforge3/envs/aehf/lib/python3.10/site-packages/grpc/_compression.py", line 15, in <module>
    from grpc._cython import cygrpc
ImportError: dlopen(/Users/shrekris/miniforge3/envs/aehf/lib/python3.10/site-packages/grpc/_cython/cygrpc.cpython-310-darwin.so, 2): Symbol not found: _CFRelease
  Referenced from: /Users/shrekris/miniforge3/envs/aehf/lib/python3.10/site-packages/grpc/_cython/cygrpc.cpython-310-darwin.so
  Expected in: flat namespace
 in /Users/shrekris/miniforge3/envs/aehf/lib/python3.10/site-packages/grpc/_cython/cygrpc.cpython-310-darwin.so

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/shrekris/miniforge3/envs/aehf/lib/python3.10/site-packages/ray/__init__.py", line 101, in <module>
    _configure_system()
  File "/Users/shrekris/miniforge3/envs/aehf/lib/python3.10/site-packages/ray/__init__.py", line 65, in _configure_system
    raise ImportError(
ImportError: Failed to import grpc on Apple Silicon. On Apple Silicon machines, try `pip uninstall grpcio; conda install grpcio`. Check out https://docs.ray.io/en/master/ray-overview/installation.html#m1-mac-apple-silicon-support for more details.
```

The user must run the command in the error output to run Ray successfully. This change switches steps 3 and 4 in the installation instructions, so Ray runs successfully the first time.

## Related issue number

Closes #35509

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - This is a documentation PR with no code changes.

### CODE REVIEW

File: doc/source/ray-overview/installation.rst [hunk 1]
```diff
@@ -288,14 +288,14 @@ Multi-node clusters are untested. To get started with local Ray development:
...
 
-#. Ensure that the ``grpcio`` package is installed via forge and **not pypi**. Grpcio currently requires special compilation flags, which pypi will _not_ correctly build with. Miniforge provides a prebuilt version of grpcio for M1 macs.
-
-   * ``pip uninstall grpcio; conda install grpcio=1.43.0 -c conda-forge``
-
 #. Install Ray as you normally would.
 
    * ``pip install ray``
 
+#. Ensure that the ``grpcio`` package is installed via forge and **not pypi**. Grpcio currently requires special compilation flags, which pypi will _not_ correctly build with. Miniforge provides a prebuilt version of grpcio for M1 macs.
```
Comment (angelinalg) [inline doc/source/ray-overview/installation.rst:295]:
```suggestion
#. Ensure that the ``grpcio`` package is installed via forge and **not pypi**. Grpcio requires special compilation flags, which pypi does _not_ correctly build with. Miniforge provides a prebuilt version of grpcio for M1 macs.
```

**Review by angelinalg — APPROVED** [review]
LGTM

**Comment by angelinalg** [comment]
Thank you for fixing this, @shrekris-anyscale!

**Comment by edoakes** [comment]
deferring to @pcmoritz because he's working on fixing the `grpcio` dependency

**Review by edoakes — APPROVED** [review]

**Pull request merged**